lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleRadioSelect/__tests__/NxCollapsibleRadioSelect.test.tsx

```diff
@@ -4,173 +4,495 @@
+            optionTooltipGenerator = jest.fn().mockReturnValue('tip'),
+            view = quickRender({ optionTooltipGenerator, tooltipModifierClass: 'customClass' }),
+            options = view.getAllByRole('menuitemradio');
 
-      expect(renderOptionNoTooltipGen({ id: 'foo', name: 'Foo' })).toHaveProp('overflowTooltip', true);
-      expect(renderOptionTooltipGen({ id: 'foo', name: 'Foo' })).toHaveProp('overflowTooltip', false);
+        await runTimers();
+        expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+        await user.hover(options[0]);
+        await runTimers();
+
+        const tooltip = screen.getByRole('tooltip');
+        expect(tooltip.querySelector('.customClass')).toBeInTheDocument();
+      });
     });
   });
 
-  describe('renderCounter prop', function () {
-    it('renders nothing if selectedId prop is not provided', function () {
-      const renderPropWrapper = getShallow().renderProp('renderCounter')();
-      expect(renderPropWrapper).toBeEmptyRender();
+  describe('filtering', function() {
+    const filterProps: Props = {
+          ...minimalProps,
+          options: [
+            {id: 'foo', name: 'Foo'},
+            {id: 'bar', name: 'Bar'},
+            {id: 'baz', name: 'Baz'},
+            {id: 'boo', name: 'Boo'}
+          ],
+          onFilterChange: () => {},
+          filterThreshold: 2
+        },
+        quickFilterRender = rtlRender(NxCollapsibleRadiosSelect, filterProps);
+
+    it('renders an input child with the textbox role', function() {
+      const view = quickFilterRender(),
+          inputEl = view.getByRole('textbox');
+
+      expect(inputEl.tagName).toBe('INPUT');
+      expect(inputEl).toBeInTheDocument();
     });
 
-    it('renders <Counter> with selected option name if selectedId prop is provided', function () {
-      const renderPropWrapper = getShallow({selectedId: 'bar'}).renderProp('renderCounter')();
-      expect(renderPropWrapper).toContainReact(<Counter>Bar</Counter>);
+    it('does not render an input when onFilterChange is not provided or filterThreshold is greater than options',
+        function() {
+          const onFilterChange = jest.fn(),
+              noOnFilterChange = quickRender({ filterThreshold: 2 }),
+              highFilterThreshold = quickRender({ onFilterChange, filterThreshold: 5 });
+
+          expect(noOnFilterChange.queryByRole('textbox')).not.toBeInTheDocument();
+          expect(highFilterThreshold.queryByRole('textbox')).not.toBeInTheDocument();
+        });
+
+    it('renders a button with an accessible name of "Clear filter"', async function() {
+      const view = quickFilterRender();
+
+      await runTimers();
+      const clearBtn = view.getByRole('button', { name: 'Clear filter' });
+
+      expect(clearBtn).toBeInTheDocument();
+      expect(clearBtn).toHaveAccessibleName('Clear filter');
+    });
+
+    it('sets the aria-controls on the input to the element id', function() {
```

Comment:
Hmm should it really be the group id or should it be the menu id?

Reply:
> Hmm should it really be the group id or should it be the menu id?

I thought it would too, but it seems as though it's controlling the entire element (unless I am misreading!)
![Screenshot 2023-01-27 at 12 42 52 PM](https://user-images.githubusercontent.com/89862727/215158702-eb9b24c9-5e9d-447d-a19d-626cdd090d9d.png)

Reply:
> I thought it would too, but it seems as though it's controlling the entire element (unless I am misreading!)

Well yes, that's what it does now.  I'm suggesting that maybe that's wrong and should be adjusted

Reply:
> Well yes, that's what it does now. I'm suggesting that maybe that's wrong and should be adjusted in the implementation
Adjusted here with an internal prop: [6761b29](https://github.com/sonatype/sonatype-react-shared-components/pull/932/commits/6761b2923571ce8552271cc98388b3f244c0ee4c)